File: #3/Categoria 4/sublist.js
const sublist = (i, f, xs) => {
  const size = xs.length
  const I = (i < 0 ? i + size : i)
  const F = (f < 0 ? f + size : f)

  const take = xs.slice();

  take.splice(F)
  take.splice(0, I)

  return take
}

console.log(sublist (1, 3, [0,1,2,3,4,5,6,7,8,9,10]))
console.log(sublist (0, 11, [0,1,2,3,4,5,6,7,8,9,10]))
console.log(sublist (2, 8, [0,1,2,3,4,5,6,7,8,9,10]))

File: #3/Categoria 4/sublist.hs
sublist i f xs = drop i' (take f' xs)
  where
    size = length xs
    i' = if i < 0 then i + size else i
    f' = if f < 0 then f + size else f

main = do
  a <- readLn :: IO Int
  b <- readLn :: IO Int
  c <- readLn :: IO [Int]
  print $ sublist a b c